python/ray/autoscaler/commands.py

```diff
@@ -95,21 +96,33 @@ def create_or_update_cluster(
         override_max_workers: Optional[int], no_restart: bool,
         restart_only: bool, yes: bool, override_cluster_name: Optional[str],
         no_config_cache: bool, log_old_style: bool, log_color: str,
+        dump_command_output: bool, use_login_shells: bool,
         verbose: int) -> None:
     """Create or updates an autoscaling Ray cluster from a config json."""
     cli_logger.old_style = log_old_style
     cli_logger.color_mode = log_color
     cli_logger.verbosity = verbose
 
-    # todo: disable by default when the command output handling PR makes it in
-    cli_logger.dump_command_output = True
+    cmd_runner.config["use_login_shells"] = use_login_shells
+    cmd_runner.config["dump_command_output"] = dump_command_output
```

Comment:
this is sort of difficult because the `commands.py` never has access to an instance of `CommandRunner`, only `updater.py` does

this means that we either need to thread this state all the way through (command -> create_or_acquire_head_node -> updater -> do_update -> commandrunner) or use a singleton/global state like here

Reply:
Ah Max I think you misread my comment. I'm not suggesting an instance of CommandRunner to exist here; I'm suggesting a cleaner way to set/access global state.